django/thunderstore/community/views/community.py

```diff
@@ -0,0 +1,85 @@
+from django.conf import settings
+from django.db.models import BigIntegerField, Count, Sum
+from django.shortcuts import redirect
+from django.views import View
+from django.views.generic.list import ListView
+
+from thunderstore.cache.cache import CacheBustCondition
+from thunderstore.cache.pagination import CachedPaginator
+from thunderstore.community.models.community_site import CommunitySite
+from thunderstore.repository.models.package import Package
+
+# Should be divisible by 4 and 3
+MODS_PER_PAGE = 24
+
+
+class CommunitiesView(ListView):
+    model = CommunitySite
+    paginate_by = MODS_PER_PAGE
+    paginator_class = CachedPaginator
+    template_name = "community/communities_list.html"
+
+    def get_base_queryset(self):
+        return self.model.objects.exclude(is_listed=False)
+
+    def get_page_title(self):
+        return "Communities"
+
+    def get_cache_vary(self):
+        return "communities"
+
+    def get_full_cache_vary(self):
+        cache_vary = self.get_cache_vary()
+        return cache_vary
+
+    def order_queryset(self, queryset):
+        return (
+            queryset.prefetch_related()
+            .annotate(
+                total_downloads=Sum(
+                    "community__package_listings__package__versions__downloads",
+                    output_field=BigIntegerField(),
+                ),
+                package_count=Count("community__package_listings"),
+            )
+            .order_by(
+                "-total_downloads",
+                "community__name",
+            )
+        )
+
+    def get_queryset(self):
+        return self.order_queryset(self.get_base_queryset())
+
+    def get_paginator(
```

Comment:
Now that I think of it, do we even need pagination here? We're just displaying top 10, and user has now way to access the next page? I guess we get caching for free with it, and it might be reusable elsewhere (but YAGNI). I honestly don't know if using it makes sense, maybe ask Mythic about it.

Reply:
I just added 10 communities, but in reality it will show all the communities we have available. I'm gonna keep the 30 as an page size limit. I'll have to test it tho